homeassistant/components/binary_sensor/alarmdecoder.py

```diff
@@ -37,13 +46,15 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
 class AlarmDecoderBinarySensor(BinarySensorDevice):
     """Representation of an AlarmDecoder binary sensor."""
 
-    def __init__(self, zone_number, zone_name, zone_type):
+    def __init__(self, zone_number, zone_name, zone_type, zone_rfid):
         """Initialize the binary_sensor."""
         self._zone_number = zone_number
         self._zone_type = zone_type
         self._state = 0
         self._name = zone_name
         self._type = zone_type
+        self._rfid = zone_rfid
+        self._rfstate = 0
```

Comment:
Initialize this to `None` instead of `0`. We should also fix `self._state` if you don't mind cleaning that up as well. It should be initialized to `None` in the constructor, set to `True`/`False` in the code, and be returned directly for `is_on`.